string_search/README.md

```diff
@@ -1,39 +1,52 @@
     * Average case complexity comparision
-* ##### Activity 3
+* #### Activity 3
     * Worst case complexity comparision
-* ##### Activity 4
+* #### Activity 4
     * Custom input by user
-* ##### References
+* #### References
 # Introduction
 #### Overview
 In this micro-byte, we will be going through the working of the two most famous string matching algorithms: Rabin Karp and KMP string matching algorithm. The speciality of both these algorithms is that they achieve linear time complexity for average case, making them far more superior to naive string matching algorithm which completes string matching in quadratic time for all cases.
 We start with understanding the working of each algorithm and then compare time taken by both algorithms in best/ average/ worst case scenarios. As an additional acitivity, readers can try out their own scenarios and compare time taken by both algorithms.
+
 #### Rabin Karp algorithm
-The Rabin Karp algorithm, named after Michael O. Rabin and Richard M. Karp, is unlike other string matching algorithms which makes use of string hashing, a technique which maps a unique number to that string. Hashing has many forms, Rabin Karp algorithm utilizes what is known as a Rolling-Hash function.
-In order to achieve string matching in linear time on an average, Rabin Karp utilizes a preprocessing step which hashes the given text and pattern. Once the hashes are generated, we only compare the hash value of the pattern to the hash value of text of the same length. Only in the case of the hash values matching will we proceed with character by character comparision. If the hash values do not match, we generate the next rolling hash for the text. There can be cases where hash values of unlike strings match, we call this spurious hits. We can avoid spurious hits and the character by character comparision all together if we choose proper parameters for the hash generating function. An example for matching DNA samples using Rabin Karp Algorithm can be found in the Rabin Karp Visualizations link of References below.
+* The algorithm is named after Michael O. Rabin and Richard M. Karp.
+* Rabin Karp Algorithm utilizes string hashing, more precisely, rolling string hashing during its preprocessing step.
+* The algorithm matches the hash value of the pattern to the rolling hash value of text. If they match, it proceeds to matching the strings character by character. If they do not match, the next rolling hash value of the text is generated and the process repeats.
+* Rabin Karp algorithm can be a victim to spurious hits - matching hash values of dissimilar text.
+* We can avoid spurious hits and the character by character comparison in Rabin Karp if we choose proper parameters for the hash generating function.
+* An example demonstrating Rabin Karp algorithm to match DNA samples can be found in the Rabin Karp Visualizations link of References.
+
+![RB GIF](https://d18l82el6cdm1i.cloudfront.net/uploads/uIPjisbiCM-bruteforce.gif)
```

Comment:
Can you cut down the length of sentences by a bit - both in this algo and next one too.
Doing this will make it quite crisp and easy to read, you can drop the history of it as well.

Reply:
Will do, The authors of the algorithm were added as people generally dont know their names.
Changes will be added in the next iteration